roles/jd-client/src/lib/mod.rs

```diff
@@ -54,110 +56,31 @@ pub static IS_NEW_TEMPLATE_HANDLED: AtomicBool = AtomicBool::new(true);
 /// switching to backup Pools in case of declared custom jobs refused by JDS (which is Pool side).
 /// As a solution of last-resort, it is able to switch to Solo Mining until new safe Pools appear
 /// in the market.
+#[derive(Clone)]
 pub struct JobDeclaratorClient {
     /// Configuration of the proxy server [`JobDeclaratorClient`] is connected to.
     config: ProxyConfig,
+    task_collector: Arc<Mutex<Vec<AbortHandle>>>,
 }
 
 impl JobDeclaratorClient {
-    pub fn new(config: ProxyConfig) -> Self {
-        Self { config }
+    pub fn new(config: ProxyConfig, task_collector: Arc<Mutex<Vec<AbortHandle>>>) -> Self {
```

Comment:
`ProxyConfig` is legacy terminology, we should take this opportunity to rename this to `JdcConfig`

Reply:
Seems like a `good first issue`?